docs/topics/auth/username.txt

```diff
@@ -0,0 +1,140 @@
+-------------------
+
+Django includes two validators:
+
+.. class:: UnicodeUsernameValidators
+    
+    Validates whether the username contains Unicode characters, in addition
+    to ``@``,``.``,``+``,``-``,and ``_``. This is the default validator.
+
+.. class:: ASCIIUsernameValidators
+
+    Validates whether the username contains ASCII letters, in addition
+    to ``@``,``.``,``+``,``-``,and ``_``. 
+
+.. class:: UsernameMinimumLengthValidator(min_length=1)
+
+.. versionadded:: 3.2
+
+    Validates whether the username meets a minimum length.
+    The minimum length can be customized with the ``min_length`` parameter.
+
+Integrating validation
+----------------------
+
+There are few functions in ``django.contrib.auth.username_validation`` that
+you can call from your own forms or other code to integrate username
+validation.
+
+.. function:: username_validators_help_texts(username_validators=None)
+
+.. versionadded:: 3.2
+
+    Returns a list of the help texts of all validators. These explain the
+    username requirements to the user.
+
+.. function:: username_validators_help_text_html(username_validators=None)
+
+.. versionadded:: 3.2
+
+    Returns an HTML string with all the text in an ``<ul>``. This is
+    helpful when adding username validation to forms, as you can pass the
+    output directly to the ``help_text`` parameter of a form field.
+
+Writing your own validator
+--------------------------
+
+If Django's built-in validators are not sufficient for your requirements,
+you can write your own username validators. Validators have a fairly small
+interface. They must must implement two methods.
+
+* ``__call__(self, username)``: validate a username. Return ``None`` if the
+  username is valid, or raise a 
+  :exc:`~django.core.exceptions.ValidationError` with an error message if the
+  username is not valid.
+* ``get_help_text(self)``: provide a help text to explain the requirements to
+  the user.
+
+Any items in the ``OPTIONS`` in :setting:``AUTH_USERNAME_VALIDATORS`` for your
+validator will be passed to the constructor. All constructor arguments should
+have a default value.
+
+Here's a  basic example of a validator, with one optional setting::
+
+    from django.core.exceptions import ValidationError
+    from django.utils.translation import gettext as _
+
+    class UsernameMinimumLengthValidator:
+        def __init__(self, min_length=1):
+            self.min_length = min_length
+
+        def __call__(self, username):
+            if len(username) < self.min_length:
+                raise ValidationError(
+                    _("This username must contain at least %(min_length)d characters."),
+                    code='invalid',
+                    params={'min_length': self.min_length},
+                )
+        
+        def get_help_text(self):
+            return _(
+                "Your username must onctain at least %(min_length)d characters."
+                "% {'min_length': self.min_length}
```

Comment:
the % formatting happens outside the `ngettext()` or `_()` in your patch, not inside